common/sc_popen.cpp

```diff
@@ -31,10 +31,19 @@
 FILE *
 sc_popen(const char *command, pid_t *pidp, const char *type)
 {
-	FILE *iop;
-	int pdes[2], pid, twoway;
 	char *argv[4];
+	argv[0] = (char *)"sh";
```

Comment:
constness is almost as strong as immutability in other languages: especially string literals may be optimised with a constand data folding optimisation pass ... so one might end up modifying data, of th string literal.

better: 
```
std::array<char*, 4> argv;
std::string sh = "/bin/sh";
argv[0] = sh.c_str();
sc_popen_argv( sh.c_str(), argv.data(), ... ); 
```

of course one has to ensure that sc_popen_argv won't deferefence the pointers after returning ... (e.g. by transferring the ownership to a thread)